smartsim/_core/launcher/colocated.py

```diff
@@ -113,26 +107,45 @@ def _build_colocated_wrapper_cmd(
         sys.executable,
         "-m",
         "smartsim._core.entrypoints.colocated",
-        "+ifname",
-        interface,
         "+lockfile",
         lockfile,
         "+db_cpus",
         str(cpus),
-        "+command",
     ]
-
+    # Add in the interface if using TCP/IP
+    interface = kwargs.get("ifname",None)
+    if interface:
+        cmd.extend(["+ifname", interface])
+    cmd.append("+command")
     # collect DB binaries and libraries from the config
     db_cmd = [CONFIG.database_exe, CONFIG.database_conf, "--loadmodule", CONFIG.redisai]
+
     # add extra redisAI configurations
     for arg, value in rai_args.items():
         if value:
             # RAI wants arguments for inference in all caps
             # ex. THREADS_PER_QUEUE=1
             db_cmd.append(f"{arg.upper()} {str(value)}")
 
-    # add port and log information
-    db_cmd.extend(["--port", str(port), "--logfile", db_log])  # usually /dev/null
+    db_cmd.extend(["--port", str(kwargs["port"])])
+
+    # Add socket and permissions for UDS
+    unix_socket = kwargs.get("unix_socket", None)
+    socket_permissions = kwargs.get("socket_permissions", None)
```

Comment:
Any reason we are reaching into kwargs here? Could these be made named kwargs with default value `None`?